### PR TITLE
Run `prettier` on JavaScript via `pre-commit`

### PR DESCRIPTION
**Pull request by adamzap**
This patch removes the global `djangoproject/static/*` exclude rule in `.pre-commit-config.yaml` to enable `prettier` to run on this project's JavaScript files. The `djangoproject/static/js/lib/` directory is now excluded in the `prettier` hook because there is no value in reformatting a minified JavaScript file.

I also added a `.prettierrc` file to the root of the project to control JavaScript formatting.

@bmispelon Is this what you had in mind as a requirement to get #1835 in?

### CODE REVIEW

**Comment by adamzap** [comment]
I added a commit to show that `pre-commit` will reformat JavaScript.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-exclude: '(^djangoproject\/static\/.*$)'
+exclude: '^djangoproject\/static\/(css\/|fonts\/|img\/|robots).*$'
```
Comment (adamzap) [inline .pre-commit-config.yaml:1]: This makes `pre-commit` only run on the `js` directory in `static`. In a series of future PRs I'd like to remove this `exclude` line and fix all of the warnings.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -19,6 +19,7 @@ repos:
           - id: debug-statements
           - id: detect-private-key
           - id: end-of-file-fixer
+            exclude: '(^djangoproject\/static\/js\/lib\/.*$)'
```
Comment (adamzap) [inline .pre-commit-config.yaml:22]: We don't need to modify minified JavaScript files to add a newline at the end.

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -46,6 +47,7 @@ repos:
       hooks:
           - id: prettier
             exclude_types: [html, json, scss]
+            exclude: '(^djangoproject\/static\/js\/lib\/.*$)'
```
Comment (adamzap) [inline .pre-commit-config.yaml:50]: We don't want to run the formatter on minified JavaScript files.

File: .prettierrc [hunk 4]
```diff
@@ -0,0 +1,11 @@
+{
+    "overrides": [
+        {
+            "files": "*.js",
+            "options": {
+                "tabWidth": 2,
+                "singleQuote": true
+            }
```
Comment (adamzap) [inline .prettierrc:8]: These options are my personal preference. Let me know if this should be modified.

https://prettier.io/docs/en/options

**Comment by adamzap** [comment]
You can see in 0091a87 that `pre-commit`/`prettier` has reformatted all relevant JavaScript according to the new rules I added.

**Review by pauloxnet — APPROVED** [review]
LGTM

**Review by bmispelon — APPROVED** [review]
This is great, I'm very happy about having a an automatically enforced standard now 🎸

**Pull request merged**